examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::{errors::HasherError, DataHasher, Discriminator, Hasher, Poseidon};
+use light_sdk::{light_accounts, verify::verify, LightDiscriminator, LightTraits};
+use light_system_program::{
+    invoke::processor::CompressedProof,
+    sdk::{
+        address::derive_address,
+        compressed_account::{
+            CompressedAccount, CompressedAccountData, PackedCompressedAccountWithMerkleContext,
+        },
+        CompressedCpiContext,
+    },
+    InstructionDataInvokeCpi, NewAddressParamsPacked, OutputCompressedAccountWithPackedContext,
+};
+use light_utils::hash_to_bn254_field_size_be;
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+
+    use light_sdk::utils::create_cpi_inputs_for_new_address;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        proof: CompressedProof,
+        new_address_params: NewAddressParamsPacked,
+        name: String,
+        rdata: RData,
+        cpi_context: Option<CompressedCpiContext>,
+    ) -> Result<()> {
+        let record = NameRecord {
+            owner: ctx.accounts.signer.key(),
+            name,
+            rdata,
+        };
+        let compressed_account = create_compressed_account(&ctx, &record, &new_address_params)?;
+
+        let signer_seed = b"cpi_signer".as_slice();
+        // let signer_seed = b"name_service".as_slice();
+        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
+        let signer_seeds = [
+            signer_seed,
+            // &ctx.accounts.signer.key.to_bytes()[..],
+            &[bump],
+        ];
+
+        let inputs = create_cpi_inputs_for_new_address(
+            proof,
+            new_address_params,
+            compressed_account,
+            &signer_seeds,
+            cpi_context,
+        );
+
+        verify(ctx, &inputs, &[&signer_seeds])?;
+
+        Ok(())
+    }
+
+    pub fn update_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, NameService<'info>>,
+        compressed_account: PackedCompressedAccountWithMerkleContext,
+        proof: CompressedProof,
+        address: [u8; 32],
+        name: String,
+        rdata: RData,
+        cpi_context: Option<CompressedCpiContext>,
+    ) -> Result<()> {
+        let record = NameRecord {
+            owner: ctx.accounts.signer.key(),
+            name,
+            rdata,
+        };
+        let new_compressed_account = compressed_account_with_address(&record, address)?;
+
+        let signer_seed = b"cpi_signer".as_slice();
+        // let signer_seed = b"name_service".as_slice();
+        let bump = Pubkey::find_program_address(&[signer_seed], &ctx.accounts.self_program.key()).1;
+        let signer_seeds = [
+            signer_seed,
+            // &ctx.accounts.signer.key.to_bytes()[..],
```

Comment:
leftovers?

Reply:
Same here - I left it, because I'm unsure about it.

I've seen in other examples that signer key was added to the seeds - I guess that makes it clear that the compressed account belongs to the program, but also to the signer of the transaction?

But I didn't get it to work.

Reply:
But OK, I will remove the commented code. Let's revisit the cpi_context and custom seeds later.